virtcontainers/container.go

```diff
@@ -1359,11 +1399,11 @@ func (c *Container) removeDrive() (err error) {
 	return nil
 }
 
-func (c *Container) attachDevices() error {
+func (c *Container) attachDevices(devices []ContainerDevice) error {
 	// there's no need to do rollback when error happens,
 	// because if attachDevices fails, container creation will fail too,
 	// and rollbackFailingContainerCreation could do all the rollbacks
-	for _, dev := range c.devices {
+	for _, dev := range devices {
```

Comment:
Might be worth adding a comment explaining why `c.devices` isn't being used as this looks slightly odd (since there are two sets of devices).

Reply:
added.